### PR TITLE
Adds description to TFE Workspaces

### PR DESCRIPTION
**Pull request by petems**
## Description

Adds description attribute to workspaces

## External links

https://www.terraform.io/docs/cloud/api/workspaces.html#request-body

### CODE REVIEW

**Review by chrisarcand — CHANGES_REQUESTED** [review]
Thanks, Peter! Just a few bits to wrap up and this looks good to me.

* `TestWorkspaceCreate` needs a line added to ensure the value you set ("bar") was set and is read properly back again. (see further along in the spec)
* A description can be updated - so a field for `WorkspaceUpdateOptions` is also necessary (along with, of course, some additions to that test as well).

File: workspace.go [hunk 1]
```diff
@@ -362,6 +366,9 @@ type WorkspaceUpdateOptions struct {
 	// API and UI.
 	Name *string `jsonapi:"attr,name,omitempty"`
 
+	// A description for the workspace.
+	Description *string `jsonapi:"attr,description,omitempty"`
```
Comment (petems) [inline workspace.go:370]: @chrisarcand Addded to WorkspaceUpdateOptions as requested 👍

File: workspace_test.go [hunk 2]
```diff
@@ -334,6 +336,7 @@ func TestWorkspacesUpdate(t *testing.T) {
 			Operations:          Bool(false),
 			QueueAllRuns:        Bool(false),
 			SpeculativeEnabled:  Bool(true),
+			Description:         String("updated description"),
```
Comment (petems) [inline workspace_test.go:339]: @chrisarcand Test for WorkspaceUpdate and checking value of description done!

**Comment by petems** [comment]
@chrisarcand Done!

PS. Random thought, but it'd be nice if this used either mocks or something like go-vcr to run the tests, it's a bit of a pain having to setup TFE creds to run testing. Added as a GH issue for the future 🦖

**Review by chrisarcand — APPROVED** [review]

**Comment by petems** [comment]
@chrisarcand Are you ok to tag a release now this has been merged?

**Comment by chrisarcand** [comment]
[Released in v0.12.1](https://github.com/hashicorp/go-tfe/releases/tag/v0.12.1)

**Pull request merged**